cmd-info.c

```diff
@@ -779,11 +779,6 @@ int command_info(int argc, char *argv[], struct opts *opts)
 	if (ret < 0)
 		pr_warn("cannot open data: %s\n", opts->dirname);
 
-	snprintf(buf, sizeof(buf), "%s/info", opts->dirname);
-
-	if (stat(buf, &statbuf) < 0)
-		return -1;
```

Comment:
@namhyung , Sorry, I'm late

I'm not sure "passing the timestamp" that you said.
But what do you think about below patch ? 

```
diff --git a/cmd-info.c b/cmd-info.c
index ba54736..137b402 100644
--- a/cmd-info.c
+++ b/cmd-info.c
@@ -42,6 +42,20 @@ static char *copy_info_str(char *src)
         return dst;
 }
 
+static int read_recorded_date(void *arg)
+{
+        struct ftrace_file_handle *handle = arg;
+        struct uftrace_info *info = &handle->info;
+        struct stat statbuf;
+
+        if (stat(handle->dirname, &statbuf) < 0)
+                return -1;
+
+        info->recorded_date = copy_info_str(ctime(&statbuf.st_mtime));
+
+        return 0;
+}
+
 static int fill_exe_name(void *arg)
 {
         struct fill_handler_arg *fha = arg;
@@ -751,6 +765,10 @@ int read_ftrace_info(uint64_t info_mask, struct ftrace_file_handle *handle)
                         return -1;
                 }
         }
+
+        if (read_recorded_date(handle) < 0)
+                pr_err("cannot get the recorded date!");
+
         return 0;
 }
 
@@ -771,7 +789,6 @@ int command_info(int argc, char *argv[], struct opts *opts)
 {
         int ret;
         char buf[PATH_MAX];
-        struct stat statbuf;
         struct ftrace_file_handle handle;
         const char *fmt = "# %-20s: %s\n";
 
@@ -779,11 +796,6 @@ int command_info(int argc, char *argv[], struct opts *opts)
         if (ret < 0)
                 pr_warn("cannot open data: %s: %m\n", opts->dirname);
 
-        snprintf(buf, sizeof(buf), "%s/info", opts->dirname);
-
-        if (stat(buf, &statbuf) < 0)
-                return -1;
-
         if (opts->print_symtab) {
                 struct symtabs symtabs = {
                         .loaded = false,
@@ -799,7 +811,7 @@ int command_info(int argc, char *argv[], struct opts *opts)
         pr_out("# system information\n");
         pr_out("# ==================\n");
         pr_out(fmt, "program version", argp_program_version);
-        pr_out("# %-20s: %s", "recorded on", ctime(&statbuf.st_mtime));
+        pr_out("# %-20s: %s\n", "recorded on", handle.info.recorded_date);
 
         if (handle.hdr.info_mask & (1UL << CMDLINE))
                 pr_out(fmt, "cmdline", handle.info.cmdline);
diff --git a/uftrace.h b/uftrace.h
index b662da3..8b9ac49 100644
--- a/uftrace.h
+++ b/uftrace.h
@@ -97,6 +97,7 @@ struct uftrace_info {
         int *tids;
         double stime;
         double utime;
+        char *recorded_date;
         long vctxsw;
         long ictxsw;
         long maxrss;

```